backend/server/routers/user.py

```diff
@@ -145,5 +145,8 @@ def update_degree_length(numYears: int, token: str = DUMMY_TOKEN):
     if diff > 0:
         user['planner']['years'] += ([{"T0": [], "T1": [], "T2": [], "T3": []}] * diff)
     else:
+        for year in user['planner']['years'][diff:]:
```

Comment:
Are there tests that need updating for this?

Reply:
check out `update_degree_length.py`

there's two really rudimentary tests;
can you also add a test for the case where the startYear is after the user's end year for `test_update_start_year`

Reply:
Updated my test. I'll fix that other test during the updateStartYear PR